solutions/devsprint-caio-santos-1/FinanceApp/Modules/ContactList/ContactListRouter.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  ContactListRouter.swift
+//  FinanceApp
+//
+//  Created by Thiago Henrique Alves Ferreira on 02/02/22.
+//
+
+import UIKit
+
+protocol ContactListRouterProtocol {
+    
+    static func createModule() -> UINavigationController
+    func navigateToNewModule()
+}
+
+final class ContactListRouter: ContactListRouterProtocol {
+    static func createModule() -> UINavigationController {
+        let viewController = ContactListViewController()
+        let navigationController = UINavigationController(rootViewController: viewController)
```

Comment:
@rebertmatheus desconheço essa orientação, aqui onde trabalho a galera faz assim, poderia por gentileza me passar um artigo ou documentação onde eu possa ver esse benchmark? fiquei curioso em aprender mais sobre.

Reply:
Bem interessante essa abordagem, mas vale ver o impacto em casos simples como este que o @rebertmatheus está fazendo. No exemplo do artigo acho que faz todo sentido aplicar essa abordagem, afinal, temos um ganho significativo de tempo, porém, é fato que o trade off é um código muito mais denso (do ponto de vista de leitura)... 

Vale analisar caso a caso, mas de qualquer forma a sugestão em si já agrega muito no compartilhamento da informação 👍

Reply:
Em um outro artigo o autor fala exatamente esse ponto que citei de analisar caso a caso:

> So that’s a quick intro to Swift’s type inference capabilities. It’s also important to point out that type inference does have a computation cost associated with it, which thankfully takes place entirely at compile-time (so it won’t effect the runtime performance of our apps), but it’s still something that can be good to keep in mind when dealing with more complex expressions. However, if we do encounter an expression that takes the compiler a long time to figure out, then we could always use one of the above techniques for specifying those types manually.


https://www.swiftbysundell.com/basics/type-inference/